GetandTake/Services/Concrete/ProductManager.cs

```diff
@@ -4,90 +4,81 @@
 using GetandTake.Models.DTOs.BaseDTO;
 using GetandTake.Models.DTOs.ListDTO;
 using GetandTake.Services.Abstract;
-using Microsoft.EntityFrameworkCore;
 
 namespace GetandTake.Services.Concrete;
 
 public class ProductManager : IProductService
 {
-
     private readonly IProductRepository _repository;
-
     private readonly IMapper _mapper;
-
     public ProductManager(IProductRepository repository, IMapper mapper)
     {
         _repository = repository;
         _mapper = mapper;
     }
-
-    public IEnumerable<ProductsDTO> GetAll()
+    public async Task CreateAsync(ProductDTO productDto)
     {
-        var products = _repository.AsNoTracking()
-            .Include(product => product.Category)
-            .Include(product => product.Supplier);
-
-        return _mapper.Map<List<ProductsDTO>>(products);
+        var product = _mapper.Map<Product>(productDto);
+        await _repository.CreateAsync(product);
     }
 
-    public IEnumerable<ProductsDTO> GetAllByCategoryId(int categoryId)
+    public async Task UpdateAsync(int productId, ProductDTO productDto)
     {
-        var products = _repository.AsNoTracking()
-            .Where(category => category.CategoryID == categoryId)
-            .Include(product => product.Category)
-            .Include(product => product.Supplier);
-
-        return _mapper.Map<List<ProductsDTO>>(products);
+        var findProduct = await _repository.GetAsync(
+            product => product.ProductID == productId);
+        if (findProduct != null)
+        {
+            var product = _mapper.Map<Product>(productDto);
+            product.ProductID = productId;
+            _repository.Update(product);
+        }
     }
 
-    public IEnumerable<ProductsDTO> GetAllBySupplierId(int supplierId)
+    public void Delete(int productId) => _repository.Delete(product => product.ProductID == productId);
```

Comment:
For lambdas it's better to wrap expressions to the next line according to good practices:
```suggestion
    public void Delete(int productId) => 
              _repository.Delete(product => product.ProductID == productId);
```